tensorflow/core/kernels/range_sampler_test.cc

```diff
@@ -157,32 +157,54 @@ TEST_F(RangeSamplerTest, FixedUnigramProbabilities) {
   Env* env = Env::Default();
   string fname = io::JoinPath(testing::TmpDir(), "vocab_file");
   TF_CHECK_OK(WriteStringToFile(env, fname, kVocabContent));
-  sampler_.reset(new FixedUnigramSampler(env, 9, fname, 0.8, 0, 1, 0));
+  FixedUnigramSampler* test_sampler = new FixedUnigramSampler(9, 0.8, 0, 1, 0);
+  TF_CHECK_OK(test_sampler->SetDistributionSampler(env, fname));
+  sampler_.reset(test_sampler);
   // 1^0.8+2^0.8+4^0.8+...+256^0.8=197.05
   for (int i = 0; i < 9; i++) {
     ASSERT_NEAR(sampler_->Probability(i), pow(2, i * 0.8) / 197.05, 1e-4);
   }
 }
+TEST_F(RangeSamplerTest, FixedUnigramNoExistingFilename) {
+  Env* env = Env::Default();
+  string fname = "NoExistingFile";
+  FixedUnigramSampler* test_sampler = new FixedUnigramSampler(9, 0.8, 0, 1, 0);
+  Status s = test_sampler->SetDistributionSampler(env, fname);
+  EXPECT_TRUE(errors::IsNotFound(s)) << s;
+}
+TEST_F(RangeSamplerTest, FixedUnigramNoMatchingRangeWeights) {
+  Env* env = Env::Default();
+  string fname = io::JoinPath(testing::TmpDir(), "vocab_file");
+  TF_CHECK_OK(WriteStringToFile(env, fname, kVocabContent));
+  FixedUnigramSampler* test_sampler = new FixedUnigramSampler(8, 0.8, 0, 1, 0);
```

Comment:
This causes a memleak which reverted the PR

CC @bhack

Reply:
Ok I suppose we don't have this coverage in the OSS CI?

Reply:
I don't think we run ASAN in OSS (builds/tests take much longer).  They are run on a cadence internally.

Reply:
Just adding `sampler_.reset(test_sampler);` (or ensuring it is already there) here should suffice. (adding it right after the creation of the FixedUnigramSampler in each case sufficed when I just tested it)

Reply:
Blackbox edit at https://github.com/tensorflow/tensorflow/pull/59096